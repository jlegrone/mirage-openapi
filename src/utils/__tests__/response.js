import Response from '../response'

test('generates response object with defaults', () => {
  const response = new Response(200, undefined, undefined)
  expect(response).toMatchSnapshot()
  expect(response.toRackResponse()).toMatchSnapshot()
})

test('generates response object with custom headers and body', () => {
  const response = new Response(204, { 'Content-Type': 'test' }, { test: true })
  expect(response).toMatchSnapshot()
  expect(response.toRackResponse()).toMatchSnapshot()
})
