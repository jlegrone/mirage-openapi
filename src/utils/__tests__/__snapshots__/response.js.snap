// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates response object with custom headers and body 1`] = `
Response {
  "code": 204,
  "data": Object {
    "test": true,
  },
  "headers": Object {
    "Content-Type": "test",
  },
}
`;

exports[`generates response object with custom headers and body 2`] = `
Array [
  204,
  Object {
    "Content-Type": "test",
  },
  Object {
    "test": true,
  },
]
`;

exports[`generates response object with defaults 1`] = `
Response {
  "code": 200,
  "data": Object {},
  "headers": Object {},
}
`;

exports[`generates response object with defaults 2`] = `
Array [
  200,
  Object {
    "Content-Type": "application/json",
  },
  Object {},
]
`;
